TestControllers/TestController.psm1

```diff
@@ -506,9 +506,40 @@ Class TestController
 			$tests = 0
 			foreach ($case in $this.SetupTypeToTestCases[$key]) {
 				$originalTestName = $case.TestName
-				for ( $testIterationCount = 1; $testIterationCount -le $this.TestIterations; $testIterationCount ++ ) {
-					if ( $this.TestIterations -ne 1 ) {
-						$case.testName = "$($originalTestName)-$testIterationCount"
+				$multiplexedTestConfigs = @()
+				$testVmSizes = @()
+
+				if ($this.OverrideVMSize) {
+					$testVmSizes = $this.OverrideVMSize.Split(",").Trim()
+				}
+
+				if ($this.TestIterations -gt 1 -or $testVmSizes.Count -gt 1) {
+					foreach($testVmSize in $testVmSizes) {
+						for ($iteration = 1; $iteration -lt $this.TestIterations + 1; $iteration++) {
+							$multiplexedTestConfig = @{
+								"TestName" = "${originalTestName}";
+								"TestVmSize" = $testVmSize
+							}
+							if ($testVmSizes.Count -gt 1) {
+								$multiplexedTestConfig["TestName"] = $multiplexedTestConfig["TestName"] + "-${testVmSize}";
+								$multiplexedTestConfig["TestVmSize"] = $testVmSize
```

Comment:
this line is not needed

Reply:
We may need another entry for eligible VM size and exist with this TestVmSize. This will be coming in new PR soon. We can keel the number of VM sizes in a single line by using this, right? @lizzha

Reply:
@lizzha vm size is required in so that the override property is set afterwards.